spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarBinderIntegrationTests.java

```diff
@@ -56,6 +56,39 @@ void basicProducerConsumerBindingEndToEnd(CapturedOutput output) {
 		}
 	}
 
+	@Test
+	void useNativeEncodingDecodingWorkAsExpected(CapturedOutput output) {
+		SpringApplication app = new SpringApplication(PiStreamConfig.class);
+		app.setWebApplicationType(WebApplicationType.NONE);
+		try (ConfigurableApplicationContext ignored = app.run(
+				"--spring.pulsar.client.service-url=" + PulsarTestContainerSupport.getPulsarBrokerUrl(),
+				"--spring.cloud.function.definition=piSupplier;piLogger",
+				"--spring.cloud.stream.bindings.piLogger-in-0.destination=piSupplier-out-0",
+				"--spring.cloud.stream.bindings.piSupplier-out-0.producer.use-native-encoding=true",
+				"--spring.cloud.stream.bindings.piLogger-in-0.consumer.use-native-decoding=true",
+				"--spring.cloud.stream.pulsar.bindings.piLogger-in-0.consumer.schema-type=FLOAT",
+				"--spring.cloud.stream.pulsar.bindings.piSupplier-out-0.producer.schema-type=FLOAT",
+				"--spring.cloud.stream.pulsar.bindings.piLogger-in-0.consumer.subscription-name=native-encoding-decoding-sub-1")) {
+			Awaitility.await().atMost(Duration.ofSeconds(10))
+					.until(() -> output.toString().contains("Hello binder: 3.14"));
+		}
+	}
+
+	@Test
+	void springCloudStreamConversionForNonTextTypes(CapturedOutput output) {
```

Comment:
This is the same as the `basicProducerConsumerBindingEndToEnd` except that it uses `Float` instead of `String` values in the functions, correct? If so, maybe we can name them similarly? It took me a sec to figure out the diff between these 3 tests etc..

Reply:
Ya, I wanted some tests with non-text payloads doing SCSt conversion.